src/OpenTelemetry/Trace/TracerProviderSdk.cs

```diff
@@ -370,38 +370,43 @@ internal bool OnShutdown(int timeoutMilliseconds)
 
     protected override void Dispose(bool disposing)
     {
-        if (!this.Disposed)
+        if (this.Disposed)
         {
-            if (disposing)
+            return;
```

Comment:
Thanks @reyang! What do you think about added deconstructor/finalizer?

Reply:
Would you elaborate a bit? What do you intend to achieve with finalizers?

Reply:
Finalizer would run only if the Dispose method does not get called, it gives base class the opportunity to finalize. Per per docs however it's recommended to use IDispose pattern. I'm on the fence!

Reply:
> Finalizer would run only if the Dispose method does not get called, it gives base class the opportunity to finalize. Per per docs however it's recommended to use IDispose pattern. I'm on the fence!

This sounds like "try to do nondeterministic things with the vague hope that it might make a buggy application feel better", I don't recommend going down this path. I think stick with IDispose is the right direction.